E2E: add test to check admin bar presence in mobile.

worldomonation opened the pull request:
#### Changes proposed in this Pull Request

This PR adds a test as per #57190 to check the presence and basic functionality of the admin bar (also called navbar).

Key changes:
- change name of some existing selectors.
- specify `:visible` pseudo-CSS.
- add handling for different mobile/desktop behavior for exiting the editor.

Details:
Changes in this PR arose from the parent issue #57190 and #51162 where on the mobile web editor no methods were provided for the user to exit the editor back to WordPress.com. 

The changes in #57234 were to persistently show the navbar in the editor view, so the user can click on `My Sites` to return to WordPress.com.

However, this means that experience of exiting the editor differs depending on the viewport size used:
- desktop: click on the site icon, then click on 'Dashboard'.
- mobile: click on My Sites.

The method `openNavSidebar` has been restructured so that it only performs its action on desktop.
Similarly, `returnToHomeDashboard` has been restructured to handle both scenarios.

#### Testing instructions

- [ ] stress test
  - [x] mobile 6924664
  - [x] desktop 6924674
- [ ] normal test
  - [x] mobile
  - [x] desktop

Closes #57190.

worldomonation committed:
Change how `openNavSidebar` and `returnToHomeDashboard` works.
- support both mobile and desktop sizes

worldomonation committed:
Update tests

worldomonation committed:
Use an array of promises intead.
Add new sped `Editor: Navbar`

worldomonation committed:
Fix lint error.

worldomonation committed:
Another lint error.

dpasque commented on packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts:
```diff
@@ -355,18 +357,43 @@ export class GutenbergEditorPage {
 		const frame = await this.getEditorFrame();
-		await frame.click( selectors.dashboardLink );
+		const navbarComponent = new NavbarComponent( this.page );
+		const actions: unknown[] = [
```
I believe we know the type here, right? It's a `Promise<any>[]` I believe!

worldomonation replied:
Ah yes I was missing the `Promise` part. Reason why I chose `unknown` is because the transpiler won't show me a warning that way :P

dpasque commented on packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts:
```diff
@@ -355,18 +357,43 @@ export class GutenbergEditorPage {
+	 * For both cases the esulting page will be the `My Home` page.
 	 */
-	async returnToDashboard(): Promise< void > {
+	async returnToHomeDashboard(): Promise< void > {
```
Optional: but I kind of feel like that function should take care of opening the nav sidebar for desktop if needed. It feels a little weird that people using this function have to include a call to `openNavSidebar` in their specs before it, but that only matters for the desktop case.

dpasque commented on test/e2e/specs/specs-playwright/wp-editor__navbar.ts:
```diff
@@ -0,0 +1,37 @@
+/**
+ * @group gutenberg
```
I think we should run this on `calypso-pr` runs, right? That navbar is pure calypso, and we override some stuff in the nav sidebar, so I think it's directly relevant to PR runs!

worldomonation replied:
Oh, I completely forgot about this one - I copy-pasted the test contents and simply forgot to update it.

dpasque reviewed (commented):
Added a few comments! The only one I feel strongly about the `calypso-pr` group inclusion!

Otherwise looks great - thanks for doing this! 👍

worldomonation committed:
Minor updates from code review.
- open the sidebar as part of the returning to home dashboard.
- change type of the action array.

dpasque reviewed (approved):
🚢
